src/overlay/PeerRecord.cpp

```diff
@@ -358,11 +362,10 @@ PeerRecord::backOff(VirtualClock& clock)
 std::chrono::seconds
 PeerRecord::computeBackoff(VirtualClock& clock)
 {
-    uint32 backoffCount = std::min<uint32>(MAX_BACKOFF_EXPONENT, mNumFailures);
+    int32 backoffCount = std::min<int32>(MAX_BACKOFF_EXPONENT, mNumFailures);
 
     auto nsecs = std::chrono::seconds(
-        std::rand() %
```

Comment:
for paranoid extra safety: backoffCount could be negative here (db error?), then value after % will be 0 and this could crash, maybe we need a check here?

Reply:
never mind, I see it fixed in 8cbe34399973cf89e81bff5dc7e0d3158942b55f